FirebaseAdmin/FirebaseAdmin/Messaging/InstanceIdClient.cs

```diff
@@ -118,8 +118,7 @@ private async Task<TopicManagementResponse> SendInstanceIdRequest(string topic,
                     throw new ArgumentException("unexpected response from topic management service");
                 }
 
-                var results = instanceIdServiceResponse.Results.Select(r => r.Error).ToList();
-                return new TopicManagementResponse(results);
+                return new TopicManagementResponse(instanceIdServiceResponse);
```

Comment:
It's probably cleaner to move the above if condition to the constructor of `TopicManagementResponse`.

PS: It seems you already have it there. Remove the redundant check from here.